CodeEdit/Features/Git/SourceControlManager.swift

```diff
@@ -95,6 +112,29 @@ final class SourceControlManager: ObservableObject {
         fileManager.notifyObservers(updatedItems: updatedStatusFor)
     }
 
+    /// Start periodic fetch with a specified interval
+    func startPeriodicFetch(interval: TimeInterval) {
+        fetchTimer?.invalidate() // Invalidate any existing timer
+        fetch()
+        fetchTimer = Timer.scheduledTimer(withTimeInterval: interval, repeats: true) { [weak self] _ in
+            self?.fetch()
+        }
+    }
+
+    /// Fetch from remote
+    func fetch() {
+        Task {
+            try await gitClient.fetchFromRemote()
```

Comment:
Should we have a notification when a fetch fails?

Reply:
I am not exactly sure how to handle this yet. We do a fetch every 10 seconds. What happens when one of those fails? Do we stop the timer when we get one failed fetch to avoid repeated alerts? When should it start back up? It will obviously fail when the user loses their internet connection. I don't necessary want to be super disruptive if they have a lapse in connection.

Reply:
What if you had the following kind of handling:

If they are offline then just show a little notification in the corner:
![Notification CodeEdit](https://github.com/CodeEditApp/CodeEdit/assets/74873375/c8d02551-8bc3-4cd1-9c42-12f3f6c2af89)
and when the person is not offline, we could do exponential backoff starting at 5 seconds and going up to 60 seconds:
d = 5*2^n

1. First attempt 5 seconds
2. Second attempt 10 seconds
3. Third attempt 20 seconds
4. Fourth attempt 40 seconds
5. Fifth attempt 60 seconds

Then instead of trying again for the sixth attempt, we can display something like this (in app as with the other case)
![Notification CodeEdit 2](https://github.com/CodeEditApp/CodeEdit/assets/74873375/342facc2-4e98-4da6-bff5-9c934d4b0b21)

Reply:
Or we simply disable the fetch timer when they lose connection and enable it again when the connection returns.

Reply:
However, we can't assume their remote requires an internet connection to get to. It could be on their local network or even on their local machine. A periodic failed fetch isn't going to necessarily hurt anything. Maybe after a set amount of failed attempts we alert the user that fetch has failed for a given timeframe.

Reply:
> Maybe after a set amount of failed attempts we alert the user that fetch has failed for a given timeframe.

Yeah this could work.

Reply:
This revision can be handled in a separate PR if necessary.